### PR TITLE
Mks 52 adjust routing

### PR DESCRIPTION
**Pull request by sergiynikon**

### CODE REVIEW

File: Serhii Nykon/NedoNet.API/Controllers/UsersController.cs [hunk 1]
```diff
@@ -22,7 +21,7 @@ public class UsersController : ControllerBase {
         }
 
         [HttpGet]
-        [Route("user/{id}")]
+        [Route("{id}")]
         public IActionResult GetUser(Guid id) {
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/Controllers/UsersController.cs]: Please set right PUT routing and parameter name, 
And why did you add GET Page endpoint to User controller?

File: Serhii Nykon/NedoNet.API/Controllers/UsersController.cs [hunk 2]
```diff
@@ -3,48 +3,56 @@
...
-        public IActionResult GetUser(Guid id) {
+        public async Task<IActionResult> GetAllUsers() {
+            try {
+                var users = await _usersService.GetAllUsersAsync();
 
-            return Ok();
+                return Ok(users);
+            }
+            catch (Exception) {
+                return StatusCode(StatusCodes.Status500InternalServerError);
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/Controllers/UsersController.cs]: Are you sure that it should be 500 status code?

File: Serhii Nykon/NedoNet.API/Controllers/UsersController.cs [hunk 3]
```diff
@@ -3,48 +3,56 @@
...
-        public IActionResult GetUsersPage([FromQuery] int page) {
-            return Ok();
+        [Route("{id}")]
+        public async Task<IActionResult> GetUser(Guid id) {
+            try {
+                var user = await _usersService.GetUserAsync(id);
+                return Ok(user);
+            }
+            catch (Exception) {
+                return StatusCode(StatusCodes.Status500InternalServerError);
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/Controllers/UsersController.cs]: It should be one of 400 series status code!

File: Serhii Nykon/NedoNet.API/Controllers/UsersController.cs [hunk 4]
```diff
@@ -3,48 +3,56 @@
...
         [HttpPost]
-        public IActionResult CreateUser() {
+        public IActionResult CreateUser([FromBody] User user) {
             return Ok();
         }
 
         [HttpPut]
         [Route("{id}")]
-        public IActionResult UpdateUser(Guid id) {
+        public IActionResult UpdateUser(Guid id, [FromBody] User user) {
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/Controllers/UsersController.cs:42]: It is better to not split the user model and Id field. But for now, it could be.

File: Serhii Nykon/NedoNet.API/Controllers/UsersController.cs (hunk 3 above)
Comment (sergiynikon) [reply Serhii Nykon/NedoNet.API/Controllers/UsersController.cs]: done

**Review by bvzayats — APPROVED** [review]
Looks nice

**Review by BohdanHamurda — APPROVED** [review]
